src/modules/schema.js

```diff
@@ -154,11 +121,42 @@ export default class Schema {
   static async getSchema(id, dockApi) {
     // TODO: requires blob module
     if (id === 'invalid-format') {
-      throw new Error(`Incorrect schema format`, dockApi);
+      throw new Error('Incorrect schema format', dockApi);
     } else if (id === 'invalid-id') {
       throw new Error(`Invalid schema id ${id}`, dockApi);
     }
 
     return {};
   }
+
+  /**
+   * Gets the JSON schema spec from given JSON. Will either return the stored JSON schema or get
+   * it using HTTP or will throw error if cannot get.
+   * @param {object} json
+   * @returns {Promise<object>}
+   */
+  static async getJSONSchemaSpec(json) {
+    const schemaKey = '$schema';
+    if (json[schemaKey]) {
+      // The URL might be 'http://json-schema.org/draft-07/schema' or 'http://json-schema.org/draft-07/schema#'
+      // In that case, the schema is already stored in the SDK as this is the latest JSON schema spec
+      if (json[schemaKey] === 'http://json-schema.org/draft-07/schema' || json[schemaKey] === 'http://json-schema.org/draft-07/schema-') {
```

Comment:
is `http://json-schema.org/draft-07/schema-` valid? comment above mentions `http://json-schema.org/draft-07/schema#`

Reply:
Good catch.